scanner/scanner.go

```diff
@@ -25,7 +25,7 @@ type Statser interface {
 // A Scanner is a zbuf.Batch source that also provides statistics.
 type Scanner interface {
 	Statser
-	Pull() (zbuf.Batch, error)
+	zbuf.Puller
```

Comment:
If we're going to do this, we should probably (a) do the same thing to proc.Proc and (b) move the verbiage in the proc.Proc doc comment regarding Pull over to zbuf.Puller.

Reply:
I can put zbuf.Puller into proc.Proc, but I think the proc.Proc comment reads fine as is.

Reply:
> I can put zbuf.Puller into proc.Proc, but I think the proc.Proc comment reads fine as is.

Actually, I take that back: I'll just undo the change to scanner.Scanner.

Reply:
What I'm getting at is that zbuf.Puller and proc.Proc each have a Pull method with identical semantics, that we should document those semantics once instead of twice, and that zbuf.Puller seems like the place to do that.

Reply:
I'm happy with leaving scanner.Scanner as is.